src/v/kafka/protocol/request_reader.h

```diff
@@ -113,6 +206,14 @@ class request_reader {
         return _parser.read_string(n);
     }
 
+    ss::sstring do_read_flex_string(uint32_t n) {
+        if (unlikely(n == 0)) {
+            /// FIXME: maybe return empty string?
+            throw std::out_of_range("Asked to read a 0 byte flex string");
```

Comment:
makes sense to me. what does upstream do here. there should probably be a known reference to this in the upstream impl.

Reply:
We actually do the same thing for regular non-flex strings (for negative lengths instead), so I just applied the same logic here:
```
    ss::sstring do_read_string(int16_t n) {
        if (unlikely(n < 0)) {
            /// FIXME: maybe return empty string?
            throw std::out_of_range("Asked to read a negative byte string");
        }
        return _parser.read_string(n);
    }
```
I think this is valid since this is expected to be a protocol violation. A flex-string of length 0 will be encoded with a value of 1 at the beginning of the buffer. 0 is only an expected value for nullable flex strings. The exception will bubble up to the `kafka::protocol` exception handler and be logged accordingly. Maybe I should remove the FIXME's then?

Reply:
what does the kafka impl in upstream do? that was the question. we can find out and write a comment too. 

i'm ok removing the fixme.

Reply:
Oh my bad, didn't see that. In any case heres a snippet i found:
`kafka/common/protocol/types/CompactArrayOf.java`
```
    public Object read(ByteBuffer buffer) {
        int n = ByteUtils.readUnsignedVarint(buffer);
        if (n == 0) {
            if (isNullable()) {
                return null;
            } else {
                throw new SchemaException("This array is not nullable.");
            }
        }
...
```
Looks like upstream throws as well

Reply:
yeah so i implemented this as conditionally calling one or another method. So we have `read_flex_string` and then `read_nullable_flex_string`. One will throw on zero, the other will return null